dm/syncer/dbconn/utils.go

```diff
@@ -96,3 +98,30 @@ func getSessionVariable(tctx *tcontext.Context, conn *DBConn, variable string) (
 	}
 	return value, nil
 }
+
+// TCPConnWithIOCounter is a wrapper of net.TCPConn with counter that accumulates
+// the bytes this connection reads/writes.
+type TCPConnWithIOCounter struct {
+	*net.TCPConn
+	c *atomic.Uint64
+}
+
+// NewTCPConnWithIOCounter creates a new TCPConnWithIOCounter.
+func NewTCPConnWithIOCounter(conn *net.TCPConn, c *atomic.Uint64) net.Conn {
+	return &TCPConnWithIOCounter{
+		TCPConn: conn,
+		c:       c,
```

Comment:
Does it require to separate read and write traffic?

Reply:
confirmed with @buchuitoudegou that we can add them together

Reply:
I think calculating the whole stuff is preferable since both read and write are billed by the AWS/GCP. But I'm worried about the network consumption collected in this way would be far larger than the data size (i.e. the total size of binlog and SQLs we write to the downstream) so that users would suspect our stats.